libsolidity/codegen/CompilerContext.cpp

```diff
@@ -423,23 +424,19 @@ void CompilerContext::appendInlineAssembly(
 	{
 		bool const isCreation = m_runtimeContext != nullptr;
 		yul::GasMeter meter(dialect, isCreation, _optimiserSettings.expectedExecutionsPerDeployment);
+		yul::Object obj;
+		obj.code = parserResult;
+		obj.analysisInfo = make_shared<yul::AsmAnalysisInfo>(analysisInfo);
 		yul::OptimiserSuite::run(
 			dialect,
 			&meter,
-			*parserResult,
-			analysisInfo,
+			obj,
 			_optimiserSettings.optimizeStackAllocation,
 			externallyUsedIdentifiers
 		);
-		analysisInfo = yul::AsmAnalysisInfo{};
-		if (!yul::AsmAnalyzer(
```

Comment:
Was this call moved somewhere else?

Reply:
The optimizer suite already updates the analysis info inside `obj` and thus this is not necessary anymore.